src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -1915,5 +1938,46 @@ public struct S
             comp = CreateCompilation(csharp);
             comp.VerifyDiagnostics();
         }
+
+        [Fact]
+        public void ReadOnlyMethod_CompoundPropertyAssignment()
+        {
+            var csharp = @"
+struct S
+{
+    int P1 { get => 123; set {} }
+    int P2 { readonly get => 123; set {} }
+    int P3 { get => 123; readonly set {} }
```

Comment:
>int P3 { get => 123; readonly set {} } [](start = 4, length = 38)

Can you also test `int P { readonly get => 123; readonly set {} }`?

Reply:
This error scenario is exercised in pre-existing tests.